flink-runtime/src/main/java/org/apache/flink/runtime/dispatcher/SchedulerNGFactoryFactory.java

```diff
@@ -22,19 +22,35 @@
 import org.apache.flink.configuration.ClusterOptions;
 import org.apache.flink.configuration.Configuration;
 import org.apache.flink.configuration.JobManagerOptions;
+import org.apache.flink.runtime.jobgraph.JobGraph;
+import org.apache.flink.runtime.jobgraph.JobType;
 import org.apache.flink.runtime.scheduler.DefaultSchedulerFactory;
 import org.apache.flink.runtime.scheduler.SchedulerNGFactory;
 import org.apache.flink.runtime.scheduler.declarative.DeclarativeSchedulerFactory;
 
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
 /** Factory for {@link SchedulerNGFactory}. */
 public final class SchedulerNGFactoryFactory {
 
+    private static final Logger LOG = LoggerFactory.getLogger(SchedulerNGFactoryFactory.class);
+
     private SchedulerNGFactoryFactory() {}
 
-    public static SchedulerNGFactory createSchedulerNGFactory(final Configuration configuration) {
-        final JobManagerOptions.SchedulerType schedulerTyp =
+    public static SchedulerNGFactory createSchedulerNGFactory(
+            final Configuration configuration, final JobGraph jobGraph) {
+        JobManagerOptions.SchedulerType schedulerType =
                 ClusterOptions.getSchedulerType(configuration);
-        switch (schedulerTyp) {
+
+        if (schedulerType == JobManagerOptions.SchedulerType.Declarative
+                && jobGraph.getJobType() == JobType.BATCH) {
+            LOG.warn(
+                    "Declarative Scheduler configured, but Batch job detected. Changing scheduler type to New Gen.");
```

Comment:
Given that we will likely consider this as a normal scenario, maybe log it on INFO?